Packs/RadwareCloudDDoSProtectionServices/Integrations/RadwareCloudDDoSProtectionServices/RadwareCloudDDoSProtectionServices.py

```diff
@@ -0,0 +1,237 @@
+            end_time(int): The end time for fetching events in milliseconds since epoch.
+            skip(int): The number of events to skip for pagination (default: 0).
+            take(int): The number of events to take for pagination (default: 700).
+        return dict[str, Any] | Response: A dictionary containing the fetched events and related metadata.
+        """
+
+        params = {
+            "criteria": [
+                {
+                    "key": "startTimestamp",
+                    "value": [start_time, None]
+                },
+                {
+                    "key": "endTimestamp",
+                    "value": [None, end_time]
+                },
+                {
+                    "key": "risk",
+                    "value": ["Info", "Low", "Medium", "High", "Critical"]
+                }
+            ],
+            "skip": skip,
+            "take": take
+        }
+        return self._http_request(
+            method="POST",
+            url_suffix="/api/sdcc/attack/core/analytics/object/vision/securityevents",
+            json_data=params,
+        )
+
+    def get_alerts(self, start_time=None, end_time=None, skip=0, take=700) -> dict[str, Any] | Response:
+        """
+        Fetches alerts from a specific time range with pagination support.
+        Args:
+            start_time(int): The start time for fetching alerts in milliseconds since epoch.
+            end_time(int): The end time for fetching alerts in milliseconds since epoch.
+            skip(int): The number of alerts to skip for pagination (default: 0).
+            take(int): The number of alerts to take for pagination (default: 700).
+        return: A dictionary containing the fetched alerts and related metadata.
+        """
+
+        params = {
+            "criteria": [
+                {
+                    "key": "timestamp",
+                    "value": [start_time, end_time]
+                },
+                {
+                    "key": "severity",
+                    "value": ["INFO", "LOW", "MEDIUM", "HIGH", "CRITICAL"]
+                }
+            ],
+            "skip": skip,
+            "take": take
+        }
+
+        return self._http_request(
+            method="POST",
+            url_suffix="/api/sdcc/infrastructure/core/analytics/object/operationalmessages/virtual",
+            json_data=params,
+        )
+
+
+def add_fields(events: list[dict], evnet_type: str | None):
+    """
+    Gets a list of events of a specific event type and adds the `_time` & `source_log_type` fields to the event.
+    Args:
+        events: A list of events.
+        evnet_type: The event type.
+    """
+    for event in events:
+        event['_time'] = event.get('timestamp')
+        event['source_log_type'] = evnet_type
+
+
+def fetch_events(client, last_run):
+    end_time = int(time.time() * 1000)
+    start_time = last_run.get('last_fetch_events', end_time-2)+1
```

Comment:
why +1? we know it can cause us to miss events

Reply:
if we don't do a plus 1 it will also return data from the last time

Reply:
Which is expected we will then do dedup of events which is better then losing events.